### PR TITLE
refactor test server in to common code for agent

### PR DESCRIPTION
**Pull request by ramaraochavali**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/cmd/pilot-agent/status/ready/probe_test.go [hunk 1]
```diff
@@ -87,7 +86,7 @@ server.state: 0`,
 
 	for _, tt := range cases {
 		t.Run(tt.name, func(t *testing.T) {
-			server := createAndStartServer(tt.stats)
+			server := testserver.CreateAndStartServer(tt.stats, "127.0.0.1:1234")
```
Comment (howardjohn) [inline pilot/cmd/pilot-agent/status/ready/probe_test.go:89]: nit: I know we don't do it today, but is it possible to use `:0` so the port is dynamic?

**Review by howardjohn — APPROVED** [review]

File: pilot/cmd/pilot-agent/status/testserver/server.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	l, err := net.Listen("tcp", address)
+	if err != nil {
+		panic("Could not create listener for test: " + err.Error())
+	}
+	server.Listener = l
+	server.Start()
+	return server
+}
+
+func createDefaultFuncMap(statsToReturn string) map[string]func(rw http.ResponseWriter, _ *http.Request) {
```
Comment (adiprerepa) [inline pilot/cmd/pilot-agent/status/testserver/server.go:46]: this is great

**Review by adiprerepa — COMMENTED** [review]
this is much better, awesome!

File: pilot/cmd/pilot-agent/status/ready/probe_test.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/cmd/pilot-agent/status/ready/probe_test.go:89]: yes we could but the caller checks stats on this port so we will have to return that some how back.

Comment (howardjohn) [reply pilot/cmd/pilot-agent/status/ready/probe_test.go:89]: couldn't you do something like `server.Addr().Port()` or equivilent?

Comment (ramaraochavali) [reply pilot/cmd/pilot-agent/status/ready/probe_test.go:89]: Yes .We could. I will change it

Comment (ramaraochavali) [reply pilot/cmd/pilot-agent/status/ready/probe_test.go:89]: https://github.com/istio/istio/pull/31901

**Pull request merged**